### ISSUE TITLE
old gcc version when installing icinga2 source code in redhat 6

### ISSUE THREAD

**Issue by Esteban2812** [Icinga/icinga2#5222]
## Expected Behavior

Get installed icinga2 in a custom path.
## Current Behavior

Installation from icinga2 source code, dowloaded zip file from this [link](https://github.com/Icinga/icinga2/archive/master.zip), failed indicating gcc version is old.

## Steps to Reproduce (for bugs)

1. Following requirements installed

- cmake installed
cmake version 2.8.12.2

- make already installed
GNU Make 3.81

- GCC 4.4 already installed
gcc (GCC) 4.4.7 20120313 (Red Hat 4.4.7-17)

- GCC 4.8 installed (even using this step output error is the same)

  - cat >/etc/yum.repos.d/devtools-2.repo <<REPO
    [testing-devtools-2-centos-\$releasever]
    name=testing 2 devtools for CentOS $releasever
    baseurl=https://people.centos.org/tru/devtools-2/6/\$basearch/RPMS
    gpgcheck=0
    REPO

  - devtoolset-2-gcc-c++installed
  - devtoolset-2-binutils installed
  - ln -sf /opt/rh/devtoolset-2/root/usr/bin/ld.bfd /opt/rh/devtoolset-2/root/usr/bin/ld 
  - ``for file in `find /opt/rh/devtoolset-2/root/usr/include/c++ -name c++config.h`; do``
    ``echo '#define _GLIBCXX__PTHREADS' >> $file``
    ``done``
  - export LD_LIBRARY_PATH=/opt/rh/devtoolset-2/root/usr/lib:$LD_LIBRARY_PATH
  - export PATH=/opt/rh/devtoolset-2/root/usr/bin:$PATH

  gcc --version
  gcc (GCC) 4.8.2 20140120 (Red Hat 4.8.2-15)

- openssl-devel installed
  rpm -qa | grep openssl-devel
  openssl-devel-1.0.1e-48.el6_8.1.i686
  openssl-devel-1.0.1e-48.el6_8.1.x86_64

- boost-devel installed
  rpm -qa | grep boost-devel
  boost-devel-1.41.0-28.el6.x86_64

- bison installed
  rpm -qa | grep bison
  bison-2.4.1-5.el6.x86_64

- flex installed
  rpm -qa | grep flex
  flex-2.5.35-9.el6.x86_64

2. installing icinga2
- mkdir -p /apps/opt/icingainstall
- cd /apps/opt/icingainstall
- wget https://github.com/Icinga/icinga2/archive/master.zip -O icinga2.zip
- unzip -uo icinga2.zip -d .
- cd icinga2-master
- mkdir -p build; cd build 
- mkdir -p /apps/opt/icingadir
- cmake .. -DCMAKE_INSTALL_PREFIX=/apps/opt/icingadir

3. Error message
```
-- Found logrotate (found version "3.7.8")
-- Boost version: 1.41.0
-- Found the following Boost libraries:
--   thread
--   system
--   program_options
--   regex
-- Could NOT find yajl (missing:  YAJL_LIBRARY YAJL_INCLUDE_DIR)
-- Could NOT find Editline (missing:  EDITLINE_LIBRARIES EDITLINE_INCLUDE_DIR)
-- Could NOT find Termcap (missing:  TERMCAP_LIBRARIES TERMCAP_INCLUDE_DIR)
CMake Error at CMakeLists.txt:250 (message):
  Your version of GCC (4.4.7) is too old for building Icinga 2 (GCC >= 4.6.0
  is required).


-- Configuring incomplete, errors occurred!
See also "/apps/opt/icingainstall/icinga2-master/build/CMakeFiles/CMakeOutput.log".
See also "/apps/opt/icingainstall/icinga2-master/build/CMakeFiles/CMakeError.log".
```
[CMakeError.log.txt](https://github.com/Icinga/icinga2/files/975287/CMakeError.log.txt)
[CMakeOutput.log.txt](https://github.com/Icinga/icinga2/files/975288/CMakeOutput.log.txt)

## Context

Installing from icinga2 from source code using steps in [link](https://github.com/Icinga/icinga2/blob/master/INSTALL.md).

## Your Environment

* Operating System and version:
Linux version 2.6.32-642.3.1.el6.x86_64 (mockbuild@x86-029.build.eng.bos.redhat.com) (gcc version 4.4.7 20120313 (Red Hat 4.4.7-17) (GCC) ) #1 SMP Sun Jun 26 18:16:44 EDT 2016

**Comment by lazyfrosch** [comment]
I build it with the official spec-file, that does `scl_enable scl enable devtoolset-2 cmake ...`

Works with the repository enabled.

As I can see in your logs it is selecting `Compiler: /usr/bin/c++`, so not the first one in PATH

**Comment by Esteban2812** [comment]
Hi @lazyfrosch,
Thanks so much for taking the time to check my issue, I would like to know if it is possible for your to provide me your full PATH variable and the file you are using when you mention "oficial spec-file"

**Comment by lazyfrosch** [comment]
Here some lines from my Jenkins test build log:

```
+ scl enable devtoolset-2 -- cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DCMAKE_BUILD_TYPE=RelWithDebInfo -DICINGA2_LTO_BUILD=ON -DCMAKE_VERBOSE_MAKEFILE=ON -DBoost_NO_BOOST_CMAKE=ON -DICINGA2_RUNDIR=/var/run -DICINGA2_USER=icinga -DICINGA2_GROUP=icinga -DICINGA2_COMMAND_GROUP=icingacmd -DBOOST_LIBRARYDIR=/usr/lib64/boost153 -DBOOST_INCLUDEDIR=/usr/include/boost153 '-DBoost_ADDITIONAL_VERSIONS='\''1.53;1.53.0'\''' -DBoost_NO_SYSTEM_PATHS=TRUE -DBUILD_TESTING=FALSE -DBoost_NO_BOOST_CMAKE=TRUE -DICINGA2_PLUGINDIR=/usr/lib64/nagios/plugins '-DCMAKE_C_FLAGS:STRING=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic ' '-DCMAKE_CXX_FLAGS:STRING=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic ' .
```

I'm doing some ccache trickery. But the main feature is toggled by using `scl enable devtoolset-2 -- cmake ......`

This sets PATH, LD_LIBRARY_PATH - I updated the INSTALL.md to reflect the only necessary calls lately...

**Comment by dnsmichi** [comment]
Those mentioned changes will be released in 2.7.0.
